src/build.rs

```diff
@@ -49,15 +51,88 @@ fn rustc_minor_version() -> Option<u32> {
     otry!(pieces.next()).parse().ok()
 }
 
-/// Ensure that `rustup` has the `wasm32-unknown-unknown` target installed for
-/// current toolchain
-pub fn rustup_add_wasm_target(step: &Step) -> Result<(), Error> {
-    let msg = format!("{}Adding the Wasm target...", emoji::TARGET);
-    PBAR.step(step, &msg);
+/// Get rustc's sysroot as a String
+fn get_rustc_sysroot() -> Result<String, FromUtf8Error> {
+    String::from_utf8(
+        Command::new("rustc")
+            .args(&["--print", "sysroot"])
+            .stdout(Stdio::piped())
+            .output()
+            .unwrap()
+            .stdout,
+    )
+}
+
+/// Checks if the wasm32-unknown-unknown is present in rustc's sysroot.
+fn is_wasm32_target_in_sysroot(sysroot: &str) -> Result<bool, Error> {
+    let wasm32_target = "wasm32-unknown-unknown";
+
+    info!("Looking for {} in {}", wasm32_target, sysroot);
+
+    let rustlib_path = &format!("{}/lib/rustlib", sysroot.replace("\n", ""));
+
+    let ls_command = Command::new("ls")
```

Comment:
Instead of using the `ls` and `grep` commands, this can probably use the native platform APIs, and I think all it needs to do is to test for `path.exists()` perhaps?

Reply:
Yes, that true. I think that's a much better solution, thanks!